File: News_Service/src/main/java/com/dunice/projectNews/services/NewsService.java
package com.dunice.projectNews.services;

import com.dunice.projectNews.dto.response.GetNewsOutResponse;
import com.dunice.projectNews.dto.request.NewsRequest;
import com.dunice.projectNews.dto.response.PageableResponse;
import java.util.List;
import java.util.Set;
import java.util.UUID;

public interface NewsService {
    PageableResponse<List<GetNewsOutResponse>> getNews(Integer page, Integer perPage);

    Long createNews(NewsRequest newsRequest);

    void putNewsById(Long id, NewsRequest newsRequest);

    void deleteNewsById(Long id);

    PageableResponse<List<GetNewsOutResponse>> getNewsFind(Integer page,
                                                           Integer perPage,
                                                           String author,
                                                           String keywords,
                                                           Set<String> tags);

    PageableResponse<List<GetNewsOutResponse>> getUserNews(UUID id, Integer page, Integer perPage);
}

File: News_Service_Kotlin/src/main/kotlin/com/dunice/advancedServiceKotlin/services/NewsService.kt
package com.dunice.advancedServiceKotlin.services

import com.dunice.advancedServiceKotlin.dto.request.NewsRequest
import com.dunice.advancedServiceKotlin.dto.response.GetNewsOutResponse
import com.dunice.advancedServiceKotlin.dto.response.PageableResponse
import java.util.*

interface NewsService {
    fun getNews(page: Int, perPage: Int?): PageableResponse<List<GetNewsOutResponse>>

    fun createNews(newsRequest: NewsRequest): Long

    fun putNewsById(id: Long, newsRequest: NewsRequest)

    fun deleteNewsById(id: Long)

    fun getNewsFind(
        page: Int,
        perPage: Int,
        author: String,
        keywords: String,
        tags: Set<String>
    ): PageableResponse<List<GetNewsOutResponse>>

    fun getUserNews(id: UUID, page: Int, perPage: Int): PageableResponse<List<GetNewsOutResponse>>
}
